core/src/kvs/ds.rs

```diff
@@ -84,7 +84,7 @@ pub struct Datastore {
 	local_live_queries: Arc<RwLock<BTreeMap<LqIndexKey, Vec<LqIndexValue>>>>,
 	// Set of tracked change feeds with associated watermarks
 	// This is updated with new/removed live queries and improves cf request performance
-	cf_watermarks: Arc<RwLock<BTreeMap<LqSelector, Versionstamp>>>,
+	cf_watermarks: Arc<Mutex<BTreeMap<LqSelector, Versionstamp>>>,
```

Comment:
Is there a particular reason for switching to a Mutex over a RwLock, other than it seems like .write() is only ever called on it?

Reply:
I need reads to be equally contended with writes. There is a segment where it is only read behaviour but the intention is to prevent parallel reads as well.